lib/commands/cookies.js

```diff
@@ -13,11 +13,18 @@ commands.getCookies = async function getCookies () {
   const cookies = await this.remote.getCookies() || {cookies: []};
   // the value is URI encoded, so decode it
   // but keep all the rest of the info intact
-  return cookies.cookies.map(function mapCookie (cookie) {
-    return Object.assign({}, cookie, {
-      value: decodeURI(cookie.value),
-    });
+  let decodedCookies = [];
+  cookies.cookies.forEach(function mapCookie (cookie) {
+    try {
+      decodedCookies.push(Object.assign({}, cookie, {
+        value: decodeURI(cookie.value),
+      }));
+    } catch (error) {
+      log.debug(`Cookie ${cookie.name} was not decoded successfully. Cookie value: ${cookie.value}`);
```

Comment:
I would rather perform value decoding separately and catch exceptions there. e.g:

```
const decodedCookieValues = cookies.cookies.map((cookie) => {
  try {
    return decodeURI(cookie.value);
  } catch (e) {
    log.debug(`Cookie ${cookie.name} was not decoded successfully. Cookie value: ${cookie.value}`);
    log.error(e);
    return undefined;
  }
});
return _.zip(cookies.cookies, decodedCookieValues)
  .map(([cookie, value]) => Object.assign({}, cookie, _.isUndefined(value) ? {} : {value}));
```

Reply:
Sounds good. I've separated as you indicated.